src/main/java/org/dependencytrack/persistence/WorkflowStateQueryManager.java

```diff
@@ -113,8 +115,8 @@ public List<WorkflowState> getAllWorkflowStatesForAToken(UUID token) {
 
     public WorkflowState getWorkflowStateByTokenAndStep(UUID token, WorkflowStep step) {
         final Query<WorkflowState> query = pm.newQuery(WorkflowState.class, "this.token == :token && this.step == step");
-        query.setParameters(token, step);
-        return query.executeUnique();
+        query.setRange(0, 1);
+        return singleResult(query.execute(token, step));
```

Comment:
Curious what the reason for this change is. Are we getting more than one result for this query?

Reply:
There is a constraint on db for uniqueness of token and step but with executeUnique i was getting an exception.

Reply:
Not required, i have converted it back to executeUnique
There is a uniqueness constraint on db on token and step so we should have only one result